### PR TITLE
Documentation refactor

### PR DESCRIPTION
**Pull request by luserx0**
### Changes:
- Moved source to to src/ directory
- Split documentation from main README into doc/ and added Debugging directives
-  Substitutes the string bmx6 with bmx7 whenever applicable
- Ported HOWTO and HOWTO.txt to README.md and prettified them
- Generic typo fixes and Markdown prettifications whenever applicable
- Add .editorconfig file at the head of root demonstrating current practises
- Fix root README.md
- Add FAQ.md doc ~~> Add FAQ Section on root README.md
- Include the BMX7 logo as the repo's banner
- Add FAQ section at the root README

#### Notes:
- The changes below contain some whitespace and empty line changes. This can be reverted immediately and will be to match the scope of the PR

### CODE REVIEW

File: doc/Plugins.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+However. specific requirements may need to be fulfilled for some plugins in order to compile correctly.
+These requirements are described in the corresponding plugin section.
+
+### Config Plugin ###
+
+#### Requirements ####
+
+uci libs are needed for the bmx7-config plugin.
+To install try (old version):
+<pre>
```
Comment (aparcar) [inline doc/Plugins.md]: using ``` or 4 whitespaces would look better

File: doc/Plugins.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+wget http://downloads.openwrt.org/sources/uci-0.7.5.tar.gz
+tar xzvf uci-0.7.5.tar.gz
+cd uci-0.7.5
+make clean all install WOPTS="-pedantic -Wall"
+sudo make install
+</pre>
+
+
+#### Compile and Install ####
```
Comment (aparcar) [inline doc/Plugins.md]: Stick to correct markdown, without the trailing hashtags

File: src/lib/bmx7_json/README.md [hunk 3]
```diff
@@ -188,19 +181,21 @@ mlc115:~# cat /var/run/bmx6/json/options
...
 }
+```
 
-
-mlc115:~# cat /var/run/bmx6/json/parameters
+```
+mlc115:~# cat /var/run/bmx7/json/parameters
 { "OPTIONS": [
-               { "name": "plugin", "INSTANCES": [ { "value": "bmx6_json.so" } ] },
+               { "name": "plugin", "INSTANCES": [ { "value": "bmx7_json.so" } ] },
```
Comment (aparcar) [inline src/lib/bmx7_json/README.md]: is it possible you "beautify" this json?

Comment (luserx0) [reply src/lib/bmx7_json/README.md]: Yup

File: doc/Plugins.md (hunk 1 above)
Comment (luserx0) [reply doc/Plugins.md]: Done

File: doc/Plugins.md (hunk 2 above)
Comment (luserx0) [reply doc/Plugins.md]: Done

File: src/lib/bmx7_json/README.md (hunk 3 above)
Comment (luserx0) [reply src/lib/bmx7_json/README.md]: Done

File: README.md [hunk 4]
```diff
@@ -98,635 +98,3 @@ You can select "luci-app-bmx7" to have a nice web interface for manage and monit
 
 Finally type "make" to build the image.
 
-## Usage (hello mesh)
```
Comment (aparcar) [inline README.md]: You need to review the ToC to link to the new files.

Comment (luserx0) [reply README.md]: Yes that's one of the TODOs concerning the README.

Comment (aparcar) [reply README.md]: So you'll update that?

Comment (luserx0) [reply README.md]: Yup

**Comment by luserx0** [comment]
Ready to be merged.

Optional future work might include: figures, graphs and per source code location README documents

**Pull request merged**